2019-06-20 12:35:31,316 ERROR: Exception on /dashboard [GET] [in /Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/flask/app.py:1780]
Traceback (most recent call last):
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/flask/app.py", line 2311, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/flask/app.py", line 1834, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/flask/app.py", line 1737, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/flask/_compat.py", line 36, in reraise
    raise value
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/flask/app.py", line 1832, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/flask/app.py", line 1818, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/aisino/Projects/DockerManager/app/main/routes.py", line 40, in dashboard
    return render_template('dashboard.html', endpoint=endpoint_info)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/flask/templating.py", line 134, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/environment.py", line 869, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/loaders.py", line 125, in load
    code = environment.compile(source, name, filename)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/environment.py", line 591, in compile
    self.handle_exception(exc_info, source_hint=source_hint)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/aisino/Projects/DockerManager/app/templates/dashboard.html", line 48, in template
    {% for k in ['stacks_num'. 'services_num', 'containers_num', 'images_num', 'volumes_num', 'networks_num'] %}
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/environment.py", line 497, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 901, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 194, in parse_for
    extra_end_rules=('name:recursive',))
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 620, in parse_tuple
    args.append(parse())
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 612, in <lambda>
    parse = lambda: self.parse_expression(with_condexpr=False)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 433, in parse_expression
    return self.parse_or()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 450, in parse_or
    left = self.parse_and()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 459, in parse_and
    left = self.parse_not()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 470, in parse_not
    return self.parse_compare()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 474, in parse_compare
    expr = self.parse_math1()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 496, in parse_math1
    left = self.parse_concat()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 507, in parse_concat
    args = [self.parse_math2()]
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 517, in parse_math2
    left = self.parse_pow()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 528, in parse_pow
    left = self.parse_unary()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 546, in parse_unary
    node = self.parse_primary()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 579, in parse_primary
    node = self.parse_list()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 649, in parse_list
    items.append(self.parse_expression())
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 432, in parse_expression
    return self.parse_condexpr()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 437, in parse_condexpr
    expr1 = self.parse_or()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 450, in parse_or
    left = self.parse_and()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 459, in parse_and
    left = self.parse_not()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 470, in parse_not
    return self.parse_compare()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 474, in parse_compare
    expr = self.parse_math1()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 496, in parse_math1
    left = self.parse_concat()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 507, in parse_concat
    args = [self.parse_math2()]
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 517, in parse_math2
    left = self.parse_pow()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 528, in parse_pow
    left = self.parse_unary()
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 547, in parse_unary
    node = self.parse_postfix(node)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 672, in parse_postfix
    node = self.parse_subscript(node)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 705, in parse_subscript
    self.fail('expected name or number', attr_token.lineno)
  File "/Users/aisino/Projects/DockerManager/venv/lib/python3.7/site-packages/jinja2/parser.py", line 59, in fail
    raise exc(msg, lineno, self.name, self.filename)
jinja2.exceptions.TemplateSyntaxError: expected name or number
2019-06-20 12:36:49,092 INFO: Aisino Docker Manager startup [in /Users/aisino/Projects/DockerManager/app/__init__.py:76]
